mlx/c/array.h

```diff
@@ -5,6 +5,7 @@
 
 #include "mlx/c/string.h"
 
+#include <cstdint>
```

Comment:
- I did not need this one on my side, curious which GCC compiler/system are you using, and where the error is located?
- this is a C header, so includes should be C, not C++; is the error popping up in `array.cpp` or in `array.h`?

Reply:
GCC 11.4: https://github.com/stemann/mlx-c/actions/runs/11910841319/job/33191101517#step:4:293

Though, I'm using GCC 10 in https://github.com/JuliaPackaging/Yggdrasil/pull/9809

Reply:
I might have messed up wrt. C vs C++ headers.

Reply:
Uh ok. Definitively need a `stdint.h` there, thanks

Reply:
To be more specific, I was using:
* Locally, in a container `mcr.microsoft.com/devcontainers/cpp:1-debian-11`: GCC 10.2 (on Debian 11).
* GitHub Actions, on `"ubuntu-latest"` runner: GCC 11.4
* BinaryBuilder env. (Julia): GCC 10.2 (cross-compile env. running on an Alpine Linux (musl) - cross-compiling to various targets)